weft/handlers.go

```diff
@@ -100,24 +102,28 @@ func MakeDirectHandler(rh DirectRequestHandler, eh ErrorHandler) http.HandlerFun
 				gz := gzip.NewWriter(w)
 				defer gz.Close()
 				w.WriteHeader(status)
-				b.WriteTo(gz)
+				n, err = b.WriteTo(gz)
 			} else {
 				w.WriteHeader(status)
-				b.WriteTo(w)
+				n, err = b.WriteTo(w)
 			}
 		}
 
 		// metrics and logging
 
+		metrics.Written(n)
+
 		metrics.Request()
 
-		switch status {
+		switch Status(err) {
```

Comment:
The Status(err) here is confusing. Since we're checking from err again which means the err value might be different to the status at line 88. However, only line 105 and 108 could change err value but it has nothing to do with the request.